### ISSUE TITLE
RecursionError on Python 3.11

### ISSUE THREAD

**Issue by cgohlke** [imageio/imageio#817]
Just a heads up. I discovered this error while testing of tifffile on Python 3.11:

```
Python 3.11.0b3 (main, Jun  1 2022, 13:29:14) [MSC v.1932 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> from imageio.v2 import imwrite
>>> imwrite('test.tif', [[0]])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Python311\Lib\site-packages\imageio\v2.py", line 238, in imwrite
    with imopen(uri, "wi", **imopen_args) as file:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python311\Lib\site-packages\imageio\core\imopen.py", line 207, in imopen
    candidate_plugin = config.plugin_class
                       ^^^^^^^^^^^^^^^^^^^
  File "C:\Python311\Lib\site-packages\imageio\config\plugins.py", line 100, in plugin_class
    module = importlib.import_module(self.module_name, self.package_name)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python311\Lib\importlib\__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1206, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1178, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1138, in _find_and_load_unlocked
  File "C:\Python311\Lib\site-packages\imageio\plugins\__init__.py", line 54, in __getattr__
    return importlib.import_module(f"imageio.plugins.{name}")
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  
<snip>

  File "C:\Python311\Lib\importlib\__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1206, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1178, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1138, in _find_and_load_unlocked
  File "C:\Python311\Lib\site-packages\imageio\plugins\__init__.py", line 54, in __getattr__
    return importlib.import_module(f"imageio.plugins.{name}")
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python311\Lib\importlib\__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1206, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1178, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1138, in _find_and_load_unlocked
  File "C:\Python311\Lib\site-packages\imageio\plugins\__init__.py", line 54, in __getattr__
    return importlib.import_module(f"imageio.plugins.{name}")
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python311\Lib\importlib\__init__.py", line 117, in import_module
    if name.startswith('.'):
       ^^^^^^^^^^^^^^^^^^^^
RecursionError: maximum recursion depth exceeded while calling a Python object
```

**Comment by cgohlke** [comment]
This seems to work for me (not exactly sure why, but it fixes the tifffile test failures):
```diff
diff --git a/imageio/plugins/__init__.py b/imageio/plugins/__init__.py
index fcdc328..2b51a34 100644
--- a/imageio/plugins/__init__.py
+++ b/imageio/plugins/__init__.py
@@ -35,6 +35,7 @@ class plugins:
     __name__ = __name__
     __loader__ = __loader__
     __file__ = __file__
+    __spec__ = __spec__

     __all__ = list(set(vars().keys()) - {"__module__", "__qualname__"})
```

**Comment by FirefoxMetzger** [comment]
Thanks for the report @cgohlke.

When we introduced lazy importing we couldn't declare `def __getattr__` on a module directly, because it wasn't supported by python 3.6. Instead, we had to use the current solution where a class replaces the module. This here seems to be a symptom of our decision, i.e., the class is being treated as a module but is missing an attribute that python 3.11 expects modules to have so things break.

Since we dropped 3.6 support with its official end of life, we should be able to refactor. That said, I'm submitting my PhD thesis this month, so I currently don't have a lot of spare time that I can spend on imageio until after the document is handed in.
